ui/common/build.gradle.kts

```diff
@@ -4,6 +4,7 @@ plugins {
     id("com.android.library")
     id("social.androiddev.code-quality")
     id("kotlin-parcelize")
+    id("com.google.osdetector") version "1.7.1"
```

Comment:
can you add plugin version to version catalog? You could reference it like this:
```kotlin
id("com.google.osdetector").version(libs.versions.osdetector.get())
```

Reply:
Unfortunately, the Gradle file doesn't recognize libs in`libs.versions.osdetector.get()`, so I added the plugin osdetector to the classpath.

Reply:
ah, that's because there's still a Gradle issue regarding that block 🙈 
We can track it here: https://youtrack.jetbrains.com/issue/KTIJ-19369 (which has been answered to be a Gradle DSL issue, tracked here: https://github.com/gradle/gradle/issues/22797)

In the meantime, you can suppress this by adding `DSL_SCOPE_VIOLATION` suppression, like so:


```kotlin
@Suppress("DSL_SCOPE_VIOLATION")
plugins {
    // other plugins here
    id("com.google.osdetector").version(libs.versions.osdetector.get())
}
```